test/spec/Feature/Query/UpdateSpec.hs

```diff
@@ -514,102 +477,50 @@ spec = do
-      get "/limited_update_items_no_pk"
-        `shouldRespondWith`
-          [json|[
-            { "id": 1, "name": "item-1" }
-          , { "id": 2, "name": "item-2" }
-          , { "id": 3, "name": "item-3" }
-          ]|]
+    it "works with views with an explicit order by unique col" $
+      verifyMutation "limited_update_items_view" tblDataBefore
+        [json|[
+          { "id": 1, "name": "item-1" }
+        , { "id": 2, "name": "updated-item" }
+        , { "id": 3, "name": "item-3" }
+        ]|]
+        $
+        request methodPatch "/limited_update_items_view?order=id&limit=1&offset=1"
+            [("Prefer", "tx=commit")]
+            [json| {"name": "updated-item"} |]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 204
+            , matchHeaders = [ matchHeaderAbsent hContentType
+                             , "Preference-Applied" <:> "tx=commit" ]
+            }
 
-      request methodPatch "/limited_update_items_no_pk?order=ctid&limit=1"
-          [("Prefer", "tx=commit")]
-          [json| {"name": "updated-item"} |]
-        `shouldRespondWith`
-          ""
-          { matchStatus  = 204
-          , matchHeaders = [ matchHeaderAbsent hContentType
-                           , "Preference-Applied" <:> "tx=commit" ]
-          }
+    it "works with views with an explicit order by composite pk" $
+      verifyMutation "limited_update_items_cpk_view" tblDataBefore
+        [json|[
+          { "id": 1, "name": "item-1" }
+        , { "id": 2, "name": "updated-item" }
+        , { "id": 3, "name": "item-3" }
+        ]|]
+        $
+        request methodPatch "/limited_update_items_cpk_view?order=id,name&limit=1&offset=1"
+            [("Prefer", "tx=commit")]
+            [json| {"name": "updated-item"} |]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 204
+            , matchHeaders = [ matchHeaderAbsent hContentType
+                             , "Preference-Applied" <:> "tx=commit" ]
+            }
 
-      get "/limited_update_items_no_pk?order=id"
-        `shouldRespondWith`
-          [json|[
-            { "id": 1, "name": "updated-item" }
-          , { "id": 2, "name": "item-2" }
-          , { "id": 3, "name": "item-3" }
-          ]|]
-
-      request methodPost "/rpc/reset_items_tables"
-        [("Prefer", "tx=commit")]
-        [json| {"tbl_name": "limited_update_items_no_pk"} |]
-        `shouldRespondWith` ""
-        { matchStatus  = 204 }
+    it "works on a table without a pk by ordering by 'ctid'" $
+      baseTable "limited_update_items_no_pk" "order=id" tblDataBefore 
+      `mutatesWith`
+      requestMutation methodPatch "/limited_update_items_no_pk?order=ctid&limit=1"
+        [json| {"name": "updated-item"} |]
+      `shouldMutateInto`
+        [json|[
+          { "id": 1, "name": "updated-item" }
+        , { "id": 2, "name": "item-2" }
+        , { "id": 3, "name": "item-3" }
+        ]|]
```

Comment:
I managed to make infix work by adding a `baseTable` function (only two parameters must be used for the infix to work this way). The `BaseTable` datatype could be used directly though.

Reply:
Looks much better! So `baseTable` takes a table name plus an `id` to ensure the order of the result. I believe the `order=` can be omitted in that case.